src/box/applier.cc

```diff
@@ -323,20 +323,14 @@ apply_nop(struct xrow_header *row)
 	return process_nop(&request);
 }
 
-/**
- * Connect to a remote host and authenticate the client.
- */
-void
-applier_connect(struct applier *applier)
+/** Common part of connection initialization. */
+static void
+applier_connection_init(struct iostream *io, const struct uri *uri,
```

Comment:
> I don't like that we pass parts of applier struct like this. 

I understand, but don't know how to do it better.
For example, io and io_ctx are also part of applier, but somewhy (because ballot watcher uses a separate io and ctx) in your variant we pass them separately. This looks inconsistent IMO.
That's why I pass all the arguments separately.